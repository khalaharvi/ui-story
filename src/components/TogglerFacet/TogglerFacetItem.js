import React from 'react';
import PropTypes from 'prop-types';

import { cx } from './TogglerFacet';

const TogglerFacetItem = ({
  item,
  checked,
  onChange,
  namespace
}) => (
  <li className={cx('list-item')}>
    <input
      id={`${namespace}-${item.attribute}`}
      type="radio"
      name={namespace}
      value={item.attribute}
      checked={checked}
      onChange={onChange}
    />
    <label htmlFor={`${namespace}-${item.attribute}`}>{item.label}</label>
  </li>
);

TogglerFacetItem.propTypes = {
  item: PropTypes.shape({
    label: PropTypes.string.isRequired,
    attribute: PropTypes.string.isRequired
  }).isRequired,
  checked: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  namespace: PropTypes.string.isRequired
};

export default TogglerFacetItem;
