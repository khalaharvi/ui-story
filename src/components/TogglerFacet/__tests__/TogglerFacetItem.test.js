import React from 'react';
import { mount, shallow } from 'enzyme';

import TogglerFacetItem from '../TogglerFacetItem';

describe('<TogglerFacetItem />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <TogglerFacetItem
        namespace="dh-toggler"
        onChange={() => null}
        checked="checked"
        item={{ attribute: 'vintage', label: 'Vintage' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when not checked', () => {
    const wrapper = shallow(
      <TogglerFacetItem
        namespace="dh-toggler"
        onChange={() => null}
        item={{ attribute: 'vintage', label: 'Vintage' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should onChange when changed', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <TogglerFacetItem
        namespace="dh-toggler"
        onChange={onChange}
        item={{ attribute: 'vintage', label: 'Vintage' }}
      />
    );

    const item = wrapper.find('li');
    expect(
      item
        .find('label')
        .text()
    ).toBe('Vintage');

    item
      .find('input[type="radio"]')
      .simulate('change', { target: { value: 'vintage' } });

    expect(onChange.mock.calls).toHaveLength(1);
    expect(onChange.mock.calls[0][0].target.value).toEqual('vintage');

    wrapper.unmount();
  });
});
