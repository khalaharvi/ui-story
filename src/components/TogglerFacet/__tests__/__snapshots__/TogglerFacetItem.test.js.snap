// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TogglerFacetItem /> renders correctly 1`] = `
<li
  className="dh-toggler-facet-list-item"
>
  <input
    checked="checked"
    id="dh-toggler-vintage"
    name="dh-toggler"
    onChange={[Function]}
    type="radio"
    value="vintage"
  />
  <label
    htmlFor="dh-toggler-vintage"
  >
    Vintage
  </label>
</li>
`;

exports[`<TogglerFacetItem /> renders correctly when not checked 1`] = `
<li
  className="dh-toggler-facet-list-item"
>
  <input
    id="dh-toggler-vintage"
    name="dh-toggler"
    onChange={[Function]}
    type="radio"
    value="vintage"
  />
  <label
    htmlFor="dh-toggler-vintage"
  >
    Vintage
  </label>
</li>
`;
